### PR TITLE
Faster onIterMap

### PR DESCRIPTION
**Pull request by MaskRay**

### CODE REVIEW

File: libr/io/io.c [hunk 1]
```diff
@@ -56,6 +56,71 @@ static int al_fd_write_at_wrap (RIO *io, int fd, ut64 addr, ut8 *buf, int len, R
 
 typedef int (*cbOnIterMap)(RIO *io, int fd, ut64 addr, ut8 *buf, int len, RIOMap *map, void *user);
 
+// If prefix_mode is true, returns the number of bytes of operated prefix; returns < 0 on error.
+// If prefix_mode is false, operates in non-stop mode and returns true iff all IO operations on overlapped maps are complete.
+static st64 on_map_skyline(RIO *io, ut64 vaddr, ut8 *buf, int len, int match_flg, cbOnIterMap op, bool prefix_mode) {
+  const RVector *skyline = &io->map_skyline;
```
Comment (radare) [inline libr/io/io.c]: Use tabs

**Comment by radare** [comment]
Cc @condret

**Comment by MaskRay** [comment]
@condret  You can add a config var to enable your bst implemention when yours is done.

Please don't stand in the way b/c it makes `aa` on `bins/mach0/swift-main` much faster.

File: libr/io/io.c (hunk 1 above)
Comment (MaskRay) [reply libr/io/io.c]: Done

**Comment by MaskRay** [comment]
N: number of skyline parts
M: number of skyline parts that overlap with `[vaddr, vaddr+len)`
Time complexity: O(log(N) + M)

**Pull request merged**